File: src/No22.js
let n = 3;
let ans = [];
let t0 = [""];
let t1 = ["()"];
ans.push(t0);
ans.push(t1);

for(let i = 2; i <= n; i ++){
    let temp = []
    for (let j = 0; j < i; j ++){
        let left = ans[j];
        let right = ans[i-j-1];
        for(let l of left){
            for(let r of right){
                let s = "(" + l + ")" + r;
                temp.push(s);
            }
        }
    }
    ans.push(temp);
}
console.log(ans[3])

File: src/No22.java
import java.util.LinkedList;
import java.util.List;

public class No22 {
    public static void main(String[] args) {
        int n = 3;
        LinkedList<LinkedList<String>> ans = new LinkedList<>();
        LinkedList<String> list0 = new LinkedList<>();
        list0.add("");
        LinkedList<String> list1 = new LinkedList<>();
        list1.add("()");
        ans.add(list0);
        ans.add(list1);
        for(int i = 2; i <= n; i ++){
            LinkedList<String> temp = new LinkedList<>();
            for(int j = 0; j < i; j ++){
                List<String> str1 = ans.get(j);
                LinkedList<String> str2 = ans.get(i-j-1);
                for(String s1:str1){
                    for(String s2:str2){
                        String tRes = "(" + s1 +")" + s2;
                        temp.add(tRes);
                    }
                }
            }
            ans.add(temp);
        }
        System.out.println(ans.get(3));
    }
}
